bigquery_etl/newtab_merino/__init__.py

```diff
@@ -0,0 +1,131 @@
+    help="Dataset in BigQuery where the source table is located.",
+)
+@click.option(
+    "--source-table", required=True, help="Name of the source table in BigQuery."
+)
+@click.option(
+    "--destination-bucket",
+    required=True,
+    help="Destination Google Cloud Storage Bucket.",
+)
+@click.option(
+    "--destination-prefix", required=True, help="Prefix of the bucket path in GCS."
+)
+@click.option(
+    "--destination-prefix", required=True, help="Prefix of the bucket path in GCS."
+)
+@click.option(
+    "--deletion-days-old",
+    required=True,
+    type=int,
+    help="Number of days after which files in GCS should be deleted.",
+)
+def export_newtab_merino_table_to_gcs(
+    source_project: str,
+    source_dataset: str,
+    source_table: str,
+    destination_bucket: str,
+    destination_prefix: str,
+    deletion_days_old: int,
+):
+    """Use bigquery client to export data from BigQuery to GCS."""
+    client = bigquery.Client(source_project)
+    error_counter = 0
+    threshold = 1
+
+    try:
+        # Generate the current timestamp
+        timestamp = datetime.utcnow().strftime("%Y%m%d%H%M")
+
+        # BigQuery does not export the proper JSON format, so we use a temp file and reformat
+        temp_file = "temp.ndjson"
+
+        job_config = bigquery.job.ExtractJobConfig(
+            destination_format=bigquery.job.DestinationFormat.NEWLINE_DELIMITED_JSON
+        )
+
+        destination_uri = f"gs://{destination_bucket}/{destination_prefix}/{temp_file}"
+
+        extract_job = client.extract_table(
+            source=f"{source_project}.{source_dataset}.{source_table}",
+            destination_uris=[destination_uri],
+            job_config=job_config,
+        )
+
+        extract_job.result()  # Waits for the job to complete.
+
+        # Verify that job was successful
+        if extract_job.state != "DONE":
+            log.error("Export failed with errors:", extract_job.errors)
+            error_counter += 1
+
+        # Initialize the storage client
+        storage_client = storage.Client()
+        bucket = storage_client.bucket(destination_bucket)
+        blob = bucket.blob(f"{destination_prefix}/{temp_file}")
+
+        # Read the temporary JSON file from GCS
+        temp_file_content = blob.download_as_text()
+
+        # Convert the content to a JSON array
+        json_array = [json.loads(line) for line in temp_file_content.splitlines()]
+        json_data = json.dumps(json_array, indent=1)
+
+        # Write the JSON array to the final destination files in GCS:
+        # 1. latest.json is a single file, that's easy to reference from Merino.
+        # 2. {timestamp}.json keeps a historical record for debugging purposes.
+        for suffix in ["latest", timestamp]:
+            final_destination_uri = f"{destination_prefix}/{suffix}.json"
+            final_blob = bucket.blob(final_destination_uri)
+            final_blob.upload_from_string(json_data, content_type="application/json")
```

Comment:
Previously, we only wrote data to `{timestamp.json`. Now we also write it to latest.json, which is easier to reference for Merino.

Reply:
please keep the timestamped data for CTR time series analysis